blockmanager.go

```diff
@@ -1725,60 +1552,26 @@ out:
 					continue
 				}
 
-				// If the block added to the main chain, then we need to
-				// update the tip locally on block manager.
 				onMainChain := !isOrphan && forkLen == 0
 				if onMainChain {
-					// Query the chain for the latest best block
-					// since the block that was processed could be
-					// on a side chain or have caused a reorg.
+					// Fetch the best block fom the state snapshot, the block processed
```

Comment:
This comment isn't correct if you look just above, you can see that this case clearly only runs when the block extended the main chain.